src/worldlogic.rs

```diff
@@ -0,0 +1,29 @@
+use crate::world::World;
+use crate::utils::Utils;
+use crate::definitions;
+use std::cell::RefCell;
+use std::rc::Rc;
+
+
+#[derive(Clone)]
+pub struct Worldlogic{
+    world:Rc<RefCell<World>>,
+}
+
+impl Worldlogic {
+    pub fn new(world: Rc<RefCell<World>>) -> Worldlogic {
+        Worldlogic {
+            world
+        }
+    }
+
+    pub fn _sandAction(x:i32,y:i32){
+        const bias = utils::random_sign();
+        return (
+            self.world.swap_tiles(x, y, x, y - 1, ["AIR", "WATER"]) ||
```

Comment:
world.swap_tiles is currently not implemented, but the vars it takes are the following: `_row1: i32, _col1: i32, _row2: i32, _col2: i32, _mask: Option<definitions::TileSet>`, the mask should be a vector TileSet though is currently incorrectly an optional single tile. [See here for reference](https://github.com/JavaRip/ant-life-optimised/blob/feature/structured-refactor/src/world.rs#L190-L192)

To create a vector of TileSet see [this example](https://github.com/JavaRip/ant-life-optimised/blob/feature/structured-refactor/src/worldgen.rs#L100-L107)